### PR TITLE
US72120 - TA86314 - fix the merge conflict for bgp_neighbor_af

### PR DESCRIPTION
**Pull request by HelenHZhang**
All beaker tests for bgp_neighbor_af passed for both nexus and xr.

### CODE REVIEW

File: tests/beaker_tests/cisco_bgp_neighbor_af/test_bgpneighboraf.rb [hunk 1]
```diff
@@ -668,25 +263,20 @@ def remove_unsupported_properties(test, platform)
...
 
-def get_dependency_manifest(platform, af, remote_as)
   extra_config = ''
   # XR requires the following before a vrf AF can be configured:
   #   1. a global router_id
   #   2. a global address family
   #   3. route_distinguisher configured on the vrf
   #   4. remote-as is required for neightbor
-  remote_as = 2 if remote_as.nil? && platform == 'ios_xr'
+  remote_as = 1 if remote_as.nil? && operating_system == 'ios_xr'
```
Comment (glennmatthews) [inline tests/beaker_tests/cisco_bgp_neighbor_af/test_bgpneighboraf.rb]: Curious - why this change from 2 to 1?

File: tests/beaker_tests/cisco_bgp_neighbor_af/test_bgpneighboraf.rb [hunk 2]
```diff
@@ -920,14 +357,17 @@ def test_harness_bgp_nbr_af(tests, id, platform)
...
-  test_harness_run(tests, :non_def_ebgp_only)
-  test_harness_run(tests, :non_def_ibgp_only)
+  extra_config = get_dependency_manifest(tests, :non_def_ebgp_only)
+  test_harness_run(tests, :non_def_ebgp_only, extra_config)
+  extra_config = get_dependency_manifest(tests, :non_def_ibgp_only)
+  test_harness_run(tests, :non_def_ibgp_only, extra_config)
 
   # -------------------------------------------------------------------
-  if platform[/n(5|6|7|9)k/]
+  if operating_system == 'iox_xr' || platform[/n(5|6|7|9)k/]
```
Comment (glennmatthews) [inline tests/beaker_tests/cisco_bgp_neighbor_af/test_bgpneighboraf.rb]: s/iox/ios/

File: tests/beaker_tests/cisco_bgp_neighbor_af/test_bgpneighboraf.rb (hunk 1 above)
Comment (HelenHZhang) [inline tests/beaker_tests/cisco_bgp_neighbor_af/test_bgpneighboraf.rb]: Before this merge, #{BgpLib::ASN} is used as ASN. After this merge we use "2" as ASN. This number is a random ASN # which is different from ASN. I thought it's easy to use "1".

File: tests/beaker_tests/cisco_bgp_neighbor_af/test_bgpneighboraf.rb (hunk 2 above)
Comment (HelenHZhang) [inline tests/beaker_tests/cisco_bgp_neighbor_af/test_bgpneighboraf.rb]: fixed it. Thanks for catching it.

Comment (glennmatthews) [inline tests/beaker_tests/cisco_bgp_neighbor_af/test_bgpneighboraf.rb]: just to confirm - this test is run and passes against XR?

Comment (HelenHZhang) [inline tests/beaker_tests/cisco_bgp_neighbor_af/test_bgpneighboraf.rb]: yes, it passed both on nexus and on XR.

**Comment by glennmatthews** [comment]
:+1:

File: tests/beaker_tests/cisco_bgp_neighbor_af/test_bgpneighboraf.rb [hunk 3]
```diff
@@ -126,7 +30,25 @@ def remove_unsupported_properties(test, platform)
   resource_name: 'cisco_bgp_neighbor_af',
 }
 
-# Test hash test cases
+# Overridden to properly handle unsupported properties.
+def unsupported_properties(_, _)
+  unpros = []
+  return unpros if operating_system == 'nexus'
+
+  unpros <<
```
Comment (chburket) [inline tests/beaker_tests/cisco_bgp_neighbor_af/test_bgpneighboraf.rb]: A nit, but you might rename ``unpros`` to ``unprops`` to match test_bgp.  Also, if you aren't appending to the array in multiple conditions, you could just define the array in a more natural way:

````
  [
    :additional_paths_receive,
    :additional_paths_send,
    :default_originate_route_map,
    ...etc...
  ]
````

File: tests/beaker_tests/cisco_bgp_neighbor_af/test_bgpneighboraf.rb [hunk 4]
```diff
@@ -730,172 +320,19 @@ def get_dependency_manifest(platform, af, remote_as)
...
-  test_harness_bgp_nbr_af(tests, 'non_default_misc_maps_part_1_l2vpn', platform)
-=======
-# TEST CASE EXECUTION
-#################################################################
 test_name "TestCase :: #{tests[:resource_name]}" do
   # -------------------------------------------------------------------
   logger.info("\n#{'-' * 60}\nSection 1. Default Property Testing")
-  test_harness_run(tests, :default)
+  extra_config = get_dependency_manifest(tests, :default)
+  test_harness_run(tests, :default, extra_config)
```
Comment (chburket) [inline tests/beaker_tests/cisco_bgp_neighbor_af/test_bgpneighboraf.rb]: Since this pattern is repeated quite a bit, consider creating a helper method, maybe something like:

````ruby
def test_harness_bgp_af_nbr(tests, id)
  extra_config = get_dependency_manifest(tests, id)
  test_harness_run(tests, id, extra_config)
end
````

File: tests/beaker_tests/cisco_bgp_neighbor_af/test_bgpneighboraf.rb (hunk 3 above)
Comment (HelenHZhang) [inline tests/beaker_tests/cisco_bgp_neighbor_af/test_bgpneighboraf.rb]: changed based on your suggestion.

File: tests/beaker_tests/cisco_bgp_neighbor_af/test_bgpneighboraf.rb (hunk 4 above)
Comment (HelenHZhang) [inline tests/beaker_tests/cisco_bgp_neighbor_af/test_bgpneighboraf.rb]: good suggestion, made change accordingly. re-ran all tests, all passed on nexus and xr.
